streams/src/main/java/org/apache/kafka/streams/kstream/internals/KGroupedTableImpl.java

```diff
@@ -68,12 +69,13 @@
     }
 
     private <T> KTable<K, T> doAggregate(final ProcessorSupplier<K, Change<V>> aggregateSupplier,
+                                         final NamedInternal named,
                                          final String functionName,
                                          final MaterializedInternal<K, T, KeyValueStore<Bytes, byte[]>> materialized) {
 
-        final String sinkName = builder.newProcessorName(KStreamImpl.SINK_NAME);
-        final String sourceName = builder.newProcessorName(KStreamImpl.SOURCE_NAME);
-        final String funcName = builder.newProcessorName(functionName);
+        final String sinkName = named.suffixWithOrElseGet("-sink", builder, KStreamImpl.SINK_NAME);
+        final String sourceName = named.suffixWithOrElseGet("-source", builder, KStreamImpl.SOURCE_NAME);
+        final String funcName = named.suffixWithOrElseGet("-sink", builder, functionName);
```

Comment:
should this have a suffix at all?  I don't think the `funcName` requires one.

If it does, at a minimum we should not use `-sink` as it's used above and doesn't accurately describe its role.

Reply:
@bbejeck ,You're right, it seems OK to remove the suffix from the function name. The user should ensure that a unique name is given to each node. So, I've removed the suffix and I also add a unit test.